Samples/Samples.AI/Prompts/WeatherPrompt.cs

```diff
@@ -0,0 +1,46 @@
+using Microsoft.Teams.AI.Annotations;
+
+namespace Samples.AI.Prompts;
+
+[Prompt]
+[Prompt.Description("Weather assistant")]
+[Prompt.Instructions("You are a helpful assistant that can help the user get the weather. First get their location, then get the weather for that location.")]
+public class WeatherPrompt
+{
+
+    [Function]
+    [Function.Description("Gets the location of the user")]
+    public string GetUserLocation()
+    {
+        var locations = new[] { "Seattle", "San Francisco", "New York" };
+        var random = new Random();
+        var location = locations[random.Next(locations.Length)];
+
+        Console.WriteLine($"[PROMPT-FUNCTION] get_user_location called, returning mock location: '{location}'");
+        return location;
+    }
+
+    [Function]
+    [Function.Description("Search for weather at a specific location")]
+    public string WeatherSearch([Param] string location)
+    {
+        Console.WriteLine($"[PROMPT-FUNCTION] weather_search called with location='{location}'");
```

Comment:
Nit: you can make use of the `IContext.Accessor` like seen in the lights sample https://github.com/microsoft/teams.net/blob/main/Samples/Samples.Lights/LightsPrompt.cs#L14C27-L14C44 to use the contexts logger, or you should also be able to inject an instance of `ILogger` in the constructor.